enricher/api/src/main/java/io/fabric8/maven/enricher/api/AbstractHealthCheckEnricher.java

```diff
@@ -0,0 +1,90 @@
+package io.fabric8.maven.enricher.api;
+
+import java.util.List;
+
+import io.fabric8.kubernetes.api.model.Container;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.KubernetesListBuilder;
+import io.fabric8.kubernetes.api.model.Probe;
+import io.fabric8.kubernetes.api.model.extensions.Deployment;
+
+/**
+ * Enriches containers with health check probes.
+ */
+public abstract class AbstractHealthCheckEnricher extends BaseEnricher {
+
+    public AbstractHealthCheckEnricher(EnricherContext buildContext, String name) {
+        super(buildContext, name);
+    }
+
+    @Override
+    public void addMissingResources(KubernetesListBuilder builder) {
+
+        List<HasMetadata> items = builder.getItems();
+        for (HasMetadata item : items) {
+            if (item instanceof Deployment) {
+                Deployment deployment = (Deployment) item;
+                Container container = getContainer(deployment);
+                if (container != null) {
+                    if (container.getReadinessProbe() == null) {
+                        Probe probe = getReadinessProbe();
+                        if (probe != null) {
+                            log.info("Adding readiness " + describe(probe));
+                            container.setReadinessProbe(probe);
+                        }
+                    }
+
+                    if (container.getLivenessProbe() == null) {
+                        Probe probe = getLivenessProbe();
+                        if (probe != null) {
+                            log.info("Adding liveness " + describe(probe));
+                            container.setLivenessProbe(probe);
+                        }
+                    }
+                }
+            }
+        }
+        builder.withItems(items);
+    }
+
+    private Container getContainer(Deployment deployment) {
+        if (deployment.getSpec() != null &&
+                deployment.getSpec().getTemplate() != null &&
+                deployment.getSpec().getTemplate().getSpec() != null &&
+                deployment.getSpec().getTemplate().getSpec().getContainers() != null &&
+                deployment.getSpec().getTemplate().getSpec().getContainers().size() > 0) {
+
+            List<Container> containers = deployment.getSpec().getTemplate().getSpec().getContainers();
+            return containers.get(containers.size() - 1); // last one
```

Comment:
Sounds good ! 

Having a single generic healthcheck features would be really, but it is by no way normative. If we encounter any problems, we should go back to specific health check enrichments. Nothing is worse then over optimised abstractions which at the end blows up anyway. 

But from my feeling I really think the image generator (or the one specifying the image) knows already how the health checks should look like. 

You are right, w.r.t to readiness probes there is no such concept in Docker (that also why d-m-p uses so called `<wait>` configs to check the startup of a container when using `docker:run` (not used in f-m-p)). But as you said, labels could fill the gap. 

It would be awesome if you could work on this. For the momemnt I'm quite happy with this PR and would like to merge it since it is very valuable and usable already. So we can work on top of this.

ok for you ?